cirq-core/cirq/ops/wait_gate.py

```diff
@@ -51,8 +51,10 @@ def __init__(
             ValueError: If the `qid_shape` provided is empty or `num_qubits` contradicts
                 `qid_shape`.
         """
-        self._duration = value.Duration(duration)
-        if not protocols.is_parameterized(self.duration) and self.duration < 0:
+        self._duration = (
+            duration if isinstance(duration, value.Duration) else value.Duration(duration)
+        )
+        if not self.duration._is_parameterized_() and self.duration.total_picos() < 0:
```

Comment:
On my Debian box the time saving is ~300ns from 2.2 to 1.9µs.

If the overall time save of this PR is ~11µs is it worth to call a dunder method of different class rather than standard `protocols.is_parameterized`?

May be we could have a similar time save by reordering the checks in the is_parameterized call.